test/test_torch.py

```diff
@@ -3889,6 +3889,44 @@ def test_comparison_ops_must_take_bool_output(self):
                        torch.logical_and, torch.logical_or, torch.logical_xor]:
                 self.assertEqual(op(torch.tensor([True]), torch.tensor([False])).dtype, torch.bool)
 
+        def test_out_comparison_ops_type_promotion_and_broadcasting(self):
+            # issue #42660
+            for op in [torch.lt, torch.le, torch.gt, torch.ge, torch.eq, torch.ne,
+                       torch.logical_and, torch.logical_or, torch.logical_xor]:
+                input1_16 = torch.ones(2, dtype=torch.bfloat16)
+                input2_32 = torch.ones(1, dtype=torch.float32)
+                output_64 = torch.zeros(1, dtype=torch.float64)
+
+                op(input1_16, input2_32, out=output_64)
+                self.assertEqual(output_64.dtype, torch.float64)
+                self.assertEqual(output_64.shape, (2,))
+
+        def test_functional_comparison_ops_return_bool_outputs(self):
+            for dtype in torch.testing.get_all_dtypes():
+                for op in [torch.eq, torch.ne,
+                        torch.logical_and, torch.logical_or, torch.logical_xor]:
+                    input1 = torch.ones(8, dtype=dtype)
+                    self.assertEqual(op(input1, input1).dtype, torch.bool)
+
+        def test_ne_eq_vectorized_treats_nan_properly(self):
```

Comment:
This won't actually call the vectorized ne, right? Because to call vectorized ne the op needs to pass a non-bool out (iirc).

Reply:
fixed, thanks :(
I originally wrote this test before making the change to avoid the temporary output tensor, so functional comparison ops would have hit the vector case then. Forgot to update the test afterwards.